packages/primitives/context-menu/src/context-menu-item-radio-group.directive.ts

```diff
@@ -0,0 +1,31 @@
+import { UniqueSelectionDispatcher } from '@angular/cdk/collections';
+import { AfterContentInit, Directive, EventEmitter, inject, Input, Output } from '@angular/core';
+
+@Directive({
+    selector: '[rdxContextMenuItemRadioGroup]',
+    standalone: true,
+    host: {
+        role: 'group'
+    },
+    providers: [{ provide: UniqueSelectionDispatcher, useClass: UniqueSelectionDispatcher }]
+})
+export class RdxContextMenuItemRadioGroupDirective<T> implements AfterContentInit {
+    private readonly selectionDispatcher = inject(UniqueSelectionDispatcher);
+
+    @Input()
+    set value(id: T | null) {
+        this._value = id;
+    }
+
+    get value(): T | null {
+        return this._value;
+    }
+
+    private _value: T | null = null;
+
+    @Output() readonly valueChange = new EventEmitter();
+
+    ngAfterContentInit(): void {
+        this.selectionDispatcher.notify(this.value as string, '');
```

Comment:
maybe?
```
if (typeof this.value === "string") {
   this.selectionDispatcher.notify(...);
}
```

Reply:
no, value has generic type

Reply:
but `selectionDispatcher.notify` waiting for a string as first arg
potential for error

Reply:
this is a problem of UniqueSelectionDispatcher :) It compares two values, and it doesn't matter what type they are. I don't know why UniqueSelectionDispatcher is written without generic types... maybe in the future we will write our own UniqueSelectionDispatcher with generic types